aten/src/ATen/SparseCsrTensorUtils.h

```diff
@@ -11,9 +11,17 @@ namespace sparse_csr {
 using SparseCsrTensor = Tensor;
 
 inline SparseCsrTensorImpl* get_sparse_csr_impl(const SparseCsrTensor& self) {
-  AT_ASSERTM(
-      self.is_sparse_csr(),
-      "_internal_get_SparseCsrTensorImpl: not a sparse CSR tensor");
+  switch (self.layout()) {
+  case kSparseCsr:
+  case kSparseCsc:
+  case kSparseBsr:
+  case kSparseBsc:
+    break;
+  default:
+    AT_ASSERTM(
+               false,
+               "_internal_get_SparseCsrTensorImpl: not a sparse CSR|CSC|BSR|BSC tensor");
```

Comment:
nit: You could also print the layout that was passed.

Reply:
Resolved via https://github.com/pytorch/pytorch/pull/75865